src/StreamFetcher.js

```diff
@@ -39,8 +39,8 @@ module.exports = class StreamFetcher {
         })
     }
 
-    _authenticate(streamId, authKey, sessionToken, operation = 'read') {
-        if (operation === 'read') {
+    _authenticate(streamId, authKey, sessionToken, operation = 'stream_subscribe') {
+        if (operation === 'stream_subscribe') {
```

Comment:
I think lines 43-47 should actually be removed! It looks like it has been an optimization to avoid doing an unnecessary permission lookup. 

Fetching the stream object requires the `stream_get` permission, not `stream_subscribe`, so that logic is now **incorrect**. This code would let users with only `stream_get` permission subscribe to streams! Maybe add a test?

Reply:
Addressed this